### ISSUE TITLE
No class found for dispatch key 'bitbucket-repository' in Prefect Version 2.8.1

### ISSUE THREAD

**Issue by sibbiii** [PrefectHQ/prefect-bitbucket#13]
Hi,

Sorry to bother again. The new 0.1.1 release works perfectly with Bitbucket Server, thanks a lot again,
but with the update of the Pipenv, Prefect was also upgraded to Version 2.8.1 from 2.8.0 which creates the error below when running:

```
(name-of-flow) PS C:\...\name-of-flow> prefect deployment build etl/transfer
_to_tableau.py:transfer_to_tableau -n useless-test -q default  --storage-block bitbucket-repository/etl-bitbucket --skip-upload
```

```
(name-of-flow) PS C:\...\name-of-flow> prefect deployment build etl/transfer
_to_abc.py:transfer_to_tableau -n useless-test -q default  --storage-block bitbucket-repository/etl-bitbucket --skip-upload
Found flow 'transfer-to-tableau'
Traceback (most recent call last):
  File "C:\Users\username\.virtualenvs\name-of-flow\lib\site-packages\prefect\cli\_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)                                                                                      
  File "C:\Users\username\.virtualenvs\name-of-flow\lib\site-packages\prefect\utilities\asyncutils.py", line 230, in coroutine_wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "C:\Users\username\.virtualenvs\name-of-flow\lib\site-packages\prefect\utilities\asyncutils.py", line 181, in run_async_in_new_loop
  File "C:\Users\username\.virtualenvs\name-of-flow\lib\site-packages\prefect\utilities\dispatch.py", line 187, in lookup_type
    raise KeyError(
KeyError: "No class found for dispatch key 'bitbucket-repository' in registry for type 'Block'."
An exception occurred.        
```
---

I spend some long time testing and what i found out is:

All works with Prefect 2.8.0 and Prefect-Bitbucket 0.1.1

Using Prefect 2.8.1 and Prefect-Bitbucket 0.1.1 leads to the error above, even when setting all up again from scratch and a worklow reduced to only a flow that does nothing and all running locally:

prefect server start
prefect config set PREFECT_API_URL="http://127.0.0.1:4200/api"
prefect block register -m prefect_bitbucket
set up the blocks in the ui
prefect deployment build filename.py:myflow -n useless-test -q default  --storage-block bitbucket-repository/etl-bitbucket --s
kip-upload

I also put in a print(registry) in dispatch.py where the error happens that leads to:

```
{'slack-webhook': <class 'prefect.blocks.notifications.SlackWebhook'>, 'ms-teams-webhook': <class 'prefect.blocks.notifications.MicrosoftTeamsWebhook'>, 'pager-duty-webhook': <class 'prefect.blocks.notifications.PagerDutyWebHook'>, 'twilio-sms': <class 'prefect.blocks.notifica
tions.TwilioSMS'>, 'opsgenie-webhook': <class 'prefect.blocks.notifications.OpsgenieWebhook'>, 'mattermost-webhook': <class 'prefect.blocks.notifications.MattermostWebhook'>, 'json': <class 'prefect.blocks.system.JSON'>, 'string': <class 'prefect.blocks.system.String'>, 'date-
time': <class 'prefect.blocks.system.DateTime'>, 'secret': <class 'prefect.blocks.system.Secret'>, 'webhook': <class 'prefect.blocks.webhook.Webhook'>, 'local-file-system': <class 'prefect.filesystems.LocalFileSystem'>, 'remote-file-system': <class 'prefect.filesystems.RemoteF
ileSystem'>, 's3': <class 'prefect.filesystems.S3'>, 'gcs': <class 'prefect.filesystems.GCS'>, 'azure': <class 'prefect.filesystems.Azure'>, 'smb': <class 'prefect.filesystems.SMB'>, 'github': <class 'prefect.filesystems.GitHub'>, 'docker-registry': <class 'prefect.infrastruct
ure.docker.DockerRegistry'>, 'docker-container': <class 'prefect.infrastructure.docker.DockerContainer'>, 'kubernetes-cluster-config': <class 'prefect.blocks.kubernetes.KubernetesClusterConfig'>, 'kubernetes-job': <class 'prefect.infrastructure.kubernetes.KubernetesJob'>, 'pro
cess': <class 'prefect.infrastructure.process.Process'>}
```

But the blocks exitst in the UI. I can work with them, and they are found as the prefect deployment build ... command needs to use the correct block name.

I'm not sure what else I can test, it looks like an incompatibility of prefect-bitbucket with the new release that renamed a lot of things. Maybe i'm doing something wrong, but why does it then work in Prefect 2.8.0 but not 2.8.1 ...

Thanks,
Sebastian

**Comment by ahuang11** [comment]
I think `KeyError: "No class found for dispatch key 'bitbucket-repository' in registry for type 'Block'."` usually indicates you don't have the package, `prefect-bitbucket`, installed in your deployment.

See https://discourse.prefect.io/t/no-class-found-for-dispatch-key-s3-bucket-in-registry-for-type-block-working-with-deployments/2373 for potential fix. Let me know if that works.

**Comment by sibbiii** [comment]
I know, but i'm running local and it is for sure is in the pipenv. I also checked the pipenv.lock and synced it. What totally puzzles me is that Prefect 2.8.0 and Prefect-Bitbucket 0.1.1 work but Prefect 2.8.1 and Prefect-Bitbucket 0.1.1 lead to this error. Hmm, thinking what i could have forgotten ...

**Comment by sibbiii** [comment]
Hi,

I upgraded again to 2.8.1 to test some things. Checked the pipenv.lock and synced again to make sure there is no issue.

First: running
```
    from prefect_bitbucket.repository import BitBucketRepository
    bitbucket_repository_block = BitBucketRepository.load("etl-bitbucket")
    print(bitbucket_repository_block.bitbucket_credentials.username)
```

runs fine. So i assume the package is installed. The error stated above stays, terminal locally in the same venv.

----

Next test:
- Prefect 2.8.0 with Bitbucket-Prefect 0.1.1 -> works
- Prefect 2.8.0 with Bitbucket-Prefect REMOVED -> same error
- Prefect 2.8.0 with Bitbucket-Prefect 0.1.1 added again -> works
- Prefect 2.8.1 with Bitbucket-Prefect 0.1.1 -> same error
- Prefect 2.8.1 with Bitbucket-Prefect REMOVED -> same error

so it looks as the package is not recognized even if installed with prefect 2.8.1 by prefect on the cli

**Comment by sibbiii** [comment]
Hi,

I continued debugging for quite some time, so I inserted in line 518 of core.py a print statement. This is where the packages seem to get loaded:

```
        print(f"{module_str} {class_name}")  # username
        return dedent(
            f"""\
        ```python
        from {module_str} import {class_name}
```

With version 2.8.0 the output is:

```
prefect.blocks.webhook Webhook
prefect.filesystems GitHub
prefect.infrastructure.docker DockerRegistry
prefect.infrastructure.docker DockerContainer
prefect.infrastructure.kubernetes KubernetesJob
prefect.infrastructure.process Process
prefect_bitbucket.repository BitBucketRepository
Found flow 'transfer-to-abc'
```

With version 2.8.1 the output is, note that BitBucketRepository is not loaded anymore before the Found flow 'transfer-to-tableau' printout

```
prefect.blocks.webhook Webhook
prefect.filesystems GitHub
prefect.infrastructure.docker DockerRegistry
prefect.infrastructure.docker DockerContainer
prefect.infrastructure.kubernetes KubernetesJob
prefect.infrastructure.process Process
Found flow 'transfer-to-abc'
```

Now I'm suck, because the code is somehow too hard to understand for me to find the issue by myself ...
Sebastian

**Comment by ahuang11** [comment]
Can you try upgrading Prefect? I think there was something commented out that broke entry points to Collections.

**Comment by sibbiii** [comment]
Thanks for the hint, fixed in 2.8.2
